pkg/network/network.go

```diff
@@ -136,6 +134,70 @@ func Render(conf *opv1alpha1.NetworkAddonsConfigSpec, manifestDir string, opensh
 	return objs, nil
 }
 
+// RenderObjsToRemove creates list of components to be removed
+func RenderObjsToRemove(prev, conf *opv1alpha1.NetworkAddonsConfigSpec, manifestDir string, openshiftNetworkConfig *osv1.Network, clusterInfo *ClusterInfo) ([]*unstructured.Unstructured, error) {
+	log.Print("starting rendering objects to delete phase")
+	objsToRemove := []*unstructured.Unstructured{}
+	o := []*unstructured.Unstructured{}
+	var err error
+
```

Comment:
Could you add an early return here? We can skip the whole function if pref == nil.